ruby/lib/cucumber/ci_environment.rb

```diff
@@ -54,13 +56,15 @@ def detect_git(ci_environment, env)
     def detect_revision(ci_environment, env)
       if env['GITHUB_EVENT_NAME'] == 'pull_request'
         raise StandardError('GITHUB_EVENT_PATH not set') unless env['GITHUB_EVENT_PATH']
-        event = JSON.parse(IO.read(env['GITHUB_EVENT_PATH']))
-        revision = event['pull_request']['head']['sha'] rescue nil
+
+        event = JSON.parse(File.read(env['GITHUB_EVENT_PATH']))
+        revision = event.dig('pull_request', 'head', 'sha')
         raise StandardError("Could not find .pull_request.head.sha in #{env['GITHUB_EVENT_PATH']}:\n#{JSON.pretty_generate(event)}") if revision.nil?
+
         return revision
       end
 
-      return evaluate(ci_environment['git']['revision'], env)
+      evaluate(ci_environment['git']['revision'], env)
```

Comment:
This can still throw errors, use `#dig` here because `#evaluate` has `nil` handling on its first line